src/Forms/GridField/GridFieldSudoMode.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace SilverStripe\Forms\GridField;
+
+use SilverStripe\Forms\FormField;
+use SilverStripe\Forms\SudoModePasswordField;
+use SilverStripe\View\ArrayData;
+use SilverStripe\View\SSViewer;
+
+/**
+ * GridFieldSudoMode warning that the managed data is protected by SudoMode which will
+ * need to be activated before the data can be edited.
+ *
+ * This is particularly important for GridFields that contain no data, which provide
+ * no other way to activate sudo mode.
+ */
+class GridFieldSudoMode extends AbstractGridFieldComponent implements GridField_HTMLProvider
+{
+    private string $sectionTitle;
+
+    private int $columnCount;
+
+    public function __construct(string $sectionTile, int $columnCount)
+    {
+        $this->sectionTitle = $sectionTile;
+        $this->columnCount = $columnCount;
+    }
+
+    public function getHTMLFragments($gridField)
+    {
+        $form = $gridField->getForm();
```

Comment:
Throw an exception if the gridfield isn't in readonly mode - someone might add the component on its own, and it doesn't make any sense on its own if the gridfield isn't readonly.

Reply:
Updated